Configure scopes to exclude

keithpitty opened the pull request:
Up until now we have assumed that Rails and some other gems e.g. Devise are being used.

This PR introduces the configuration option to exclude keys that are scopes with a common prefix e.g. `:activerecord`.

In `I18n::Hygiene::LocaleTranslations` the `non_tc_scopes` and `scopes_from_*` methods which are currently used to exclude translation keys from those that are checked will no longer be hard-coded. Rather, client applications such as TC will need to be configured to exclude these scopes.

keithpitty committed:
Configure scopes to exclude

We need this option so that clients of the gem can configure which
scopes to exclude rather than them being hard-coded, which is a hangover
from the Rake tasks originally developed for TC.

keithpitty committed:
Use scopes_to_exclude configuration

Now that we can configure which scopes to exclude, we remove the old
hard-coded non_tc_scopes.

keithpitty commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[i18n-hygiene] Deal with Rails etc keys_to_skip](https://trello.com/c/53G1004e/7813-i18n-hygiene-deal-with-rails-etc-keystoskip)

nickbrowne commented on lib/i18n/hygiene/wrapper.rb:
```diff
@@ -8,13 +8,18 @@ module Hygiene
     class Wrapper
 
-      def initialize(keys_to_skip: [], locales: ::I18n.available_locales)
+      def initialize(keys_to_skip: [], scopes_to_exclude: [], locales: ::I18n.available_locales)
```
Should we use consistent terminology across `keys_to_skip` and `scopes_to_exclude`? They are doing similar things.

I think I like exclude better for what it's worth.

keithpitty replied:
OK, I'll change `keys_to_skip` to `keys_to_exclude`.

keithpitty committed:
Rename for consistency

We rename :keys_to_skip to :keys_to_exclude so that it is consistent
with the new :scopes_to_exclude config option.

keithpitty commented:
Rebased.

keithpitty committed:
Pass scopes_to_exclude config to other checks

keithpitty commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[i18n-hygiene] Configure TC to exclude scopes from check](https://trello.com/c/G7FZs7vX/7866-i18n-hygiene-configure-tc-to-exclude-scopes-from-check)

nickbrowne commented:
:+1:
